pkg/logs/events.go

```diff
@@ -147,6 +149,8 @@ func (ls *LogsService) handleStart(ctx context.Context) func(msg *nats.Msg) {
 // handleStop will handle stop event and stop logs consumers, also clean consumers state
 func (ls *LogsService) handleStop(ctx context.Context) func(msg *nats.Msg) {
 	return func(msg *nats.Msg) {
+		ls.log.Debugw("got stop event")
+		time.Sleep(StopWaitTime)
```

Comment:
shouldn't it be ticker with context.done check?

Reply:
looks a bit hacky

Reply:
Yep the issue is that it's all async - and this event can be handled before first message even arrive - so I need to deal somehow with it. 

Log is in consumer which I must remove from memory when not needed anymore. 
So there are events:
- START - triggers messages handling (need to know ID of execution to start consumer) 
- STOP (cleaning)

Reply:
passed ctx handling

Reply:
reminds me tcp package flow, they can also come async in a mixed form and must be reordered. Ideally, would be great to check, that log server received all messages in the stream with order number less that the stop message number, this way it will be safe to stop it. but I don't know if log server tracks it

Reply:
nope - but in general good idea - but will be hard to accomplish as assuming logs can be send from different places (e.g. job sidecar, or even test scheuduler in case of some init errors, so there will be no single counter where we can count lines :/)